### DISCUSSION TITLE
eac3 support

### DISCUSSION THREAD

**Discussion by b-wille** [bluenviron/mediamtx#3936] [Questions]
### Question

Hello,

Thank you very much for the great work.

HLS actually also supports eac3, see below.

Why doesn't mediamtx actually support eac3? If I convert the eac3 audio track to ac3, then it also works to stream with mediamtx.

Another question regarding ffmpeg location. Can I set somewhere which ffmpeg should be used?

I would be very happy to receive an answer.
Thanks

Greetings
###############################################################################
mediamtx v1.9.3
ffmpeg 6.1.2

Input #0, hls, from 'http://127.0.0.1/xyz.m3u8':
  Duration: N/A, start: 1, bitrate: N/A
  Program 0
    Metadata:
      variant_bitrate : 123456789
  Stream #0:0(deu): Audio: eac3 (ec-3 / 0x332D6365), 48000 Hz, 5.1(side), fltp, 256 kb/s (default)
    Metadata:
      variant_bitrate : 123456789
      compatible_brands: 
      handler_name    : SoundHandler
      comment         : speech
    Side data:
      audio service type: main
  Stream #0:1(deu): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp (default)
    Metadata:
      variant_bitrate : 123456789
      compatible_brands: 
      handler_name    : SoundHandler
      comment         : speech
  Stream #0:2(eng): Video: h264 (Main) (avc1 / 0x31637661), yuv420p, 1920x1080 [SAR 1:1 DAR 16:9], 50 fps, 50 tbr, 1k tbn (default)
    Metadata:
      variant_bitrate : 123456789
      handler_name    : VideoHandler
      compatible_brands: 
2024/11/05 18:12:38 INF [SRT] [conn 127.0.0.1:43653] opened
Output #0, mpegts, to 'srt://localhost:8890?streamid=publish:xyz&pkt_size=1316':
  Metadata:
    encoder         : Lavf60.16.100
  Stream #0:0(deu): Audio: eac3 (ec-3 / 0x332D6365), 48000 Hz, 5.1(side), fltp, 256 kb/s (default)
    Metadata:
      variant_bitrate : 123456789
      compatible_brands: 
      handler_name    : SoundHandler
      comment         : speech
    Side data:
      audio service type: main
  Stream #0:1(deu): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp (default)
    Metadata:
      variant_bitrate : 123456789
      compatible_brands: 
      handler_name    : SoundHandler
      comment         : speech
  Stream #0:2(eng): Video: h264 (Main) (avc1 / 0x31637661), yuv420p, 1920x1080 [SAR 1:1 DAR 16:9], q=2-31, 50 fps, 50 tbr, 90k tbn (default)
    Metadata:
      variant_bitrate : 123456789
      handler_name    : VideoHandler
      compatible_brands: 
Stream mapping:
  Stream #0:0 -> #0:0 (copy)
  Stream #0:1 -> #0:1 (copy)
  Stream #0:2 -> #0:2 (copy)
Press [q] to stop, [?] for help
[https @ 0x2b7f8380] Opening 'http://127.0.0.1/xyz/1.ts' for reading
2024/11/05 18:12:38 WAR [SRT] [conn 127.0.0.1:43653] skipping track 1 (unsupported codec)
2024/11/05 18:12:38 INF [SRT] [conn 127.0.0.1:43653] is publishing to path 'xyz', 2 tracks (MPEG-4 Audio, H264)
2024/11/05 18:12:38 INF [HLS] [muxer xyz] is converting into HLS, 2 tracks (MPEG-4 Audio, H264)

**Comment by aler9** [comment]
Hello, the reason why we don't support AC-3 or EAC-3 tracks in HLS is because browser support is extremely limited. In particular, they are supported by Safari only, there is no Chrome or Firefox support, and situation will stay the same due to licensing issues.

Currently we prefer not to support protocol/codecs combinations with limited player support (even if implementing some of these would be extremely easy) otherwise we would cause problems to people that might start using the server only to find out that their chosen codec cannot be played.

An example of this situation is the HLS+H265 combination: we chose to implement it because there was great demand, but browser support is still around 60% (basically it's limited to Windows+Chrome+GPU, iOS and macOS+GPU) and the result is that there are many people complaining about it. Even if we cannot do anything about it.
